gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -0,0 +1,208 @@
+package com.google.cloud.hadoop.gcsio.integration;
+
+import static com.google.cloud.hadoop.gcsio.integration.GoogleCloudStorageTestHelper.assertObjectContent;
+import static com.google.cloud.hadoop.gcsio.integration.GoogleCloudStorageTestHelper.writeObject;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorage;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageImpl;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageItemInfo;
+import com.google.cloud.hadoop.gcsio.ListObjectOptions;
+import com.google.cloud.hadoop.gcsio.StorageResourceId;
+import com.google.cloud.hadoop.gcsio.ThrottledGoogleCloudStorage;
+import com.google.cloud.hadoop.gcsio.ThrottledGoogleCloudStorage.StorageOperation;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.Lists;
+import com.google.common.flogger.GoogleLogger;
+import com.google.common.util.concurrent.RateLimiter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.SeekableByteChannel;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class GoogleCloudStorageGrpcIntegrationTest {
+
+  // Prefix this name with the prefix used in other gcs io integrate tests once it's whitelisted by
+  // GCS to access gRPC API.
+  private static final String BUCKET_NAME = "gcs-grpc-team-dataproc-it";
+
+  private static final GoogleCloudStorage rawStorage = getRawStorage();
```

Comment:
I think it will make tests more flexible and independent if each test will have its own instance of the `GoogleCloudStorage` - overhead of creating instance per test should be negligible.

Reply:
hmm i'm a bit confused. So to create bucket only once as you suggested we need a storage object and it needs to be static due to the @Beforeclass limitation. Are you suggesting we create another per-test rawstorage object in a e.g. @Before method?

Reply:
Yes, or we can just call static `createGoogleCloudStorage()` method in each test method directly - there little value in assigning this object to class field.

Reply:
Done.